addons/sale_loyalty/models/sale_order.py

```diff
@@ -567,7 +568,7 @@ def _get_claimable_rewards(self, forced_coupons=None):
             for reward in coupon.program_id.reward_ids:
                 if reward.is_global_discount and global_discount_reward and global_discount_reward.discount >= reward.discount:
                     continue
-                if reward.reward_type == 'discount' and total_is_zero:
+                if reward.reward_type == 'discount' and (total_is_zero or reward in self.order_line.reward_id):
```

Comment:
@Williambraecky Hey, I'm not sure why you did these changes. It's making this test fail https://github.com/odoo/odoo/blob/1d92a264a666e2c6bdf4705502bd58ff6d08cf7b/addons/sale_loyalty/tests/test_pay_with_gift_card.py#L53

Reply:
It was necessary because the tour actually showed a flaw in the auto claim.
Yes I noticed but didn't have the time to check yet. Will fix it

Reply:
@robinengels Hopefully fixed.
For some reason all tours failed on my local database.